packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -0,0 +1,352 @@
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return _GenericPageTransition(routeAnimation: animation, child: child);
+  }
+}
+
+/// Used by [PageTransitionsTheme] to define a vertical [MaterialPageRoute] page
+/// transition animation that looks like the default page transition
+/// used on Android P.
+///
+/// The [platform] for this builder [TargetPlatform.android].
+///
+/// See also:
+///  * [GenericPageTransitionsBuilder], which defines a default page transition.
+///  * [CupertinoPageTransitionsBuilder], which defines a horizontal page
+///    transition that matches nativer iOS page transitions.
+class MountainViewPageTransitionsBuilder extends PageTransitionsBuilder {
+  const MountainViewPageTransitionsBuilder() : super(platform: TargetPlatform.android);
+
+  @override
+  Widget buildTransitions<T>(
+    PageRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return _MountainViewPageTransition(
+      animation: animation,
+      secondaryAnimation: secondaryAnimation,
+      child: child,
+    );
+  }
+}
+
+/// Used by [PageTransitionsTheme] to define a horizontal [MaterialPageRoute]
+/// page transition animation that matches native iOS page transitions.
+///
+/// The [platform] for this builder [TargetPlatform.iOS].
+///
+/// See also:
+///
+///  * [GenericPageTransitionsBuilder], which defines a default page transition.
+///  * [MountainViewPageTransitionsBuilder], which defines a page transition
+///    that's similar to the one provided by Android P.
+///  * [GenericPageTransitionsBuilder], which defines a default page transition.
+class CupertinoPageTransitionsBuilder extends PageTransitionsBuilder {
+  const CupertinoPageTransitionsBuilder() : super(platform: TargetPlatform.iOS);
+
+  @override
+  Widget buildTransitions<T>(
+    PageRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return CupertinoPageRoute.buildPageTransitions<T>(route, context, animation, secondaryAnimation, child);
+  }
+}
+
+/// Defines the page transition animations used by [MaterialPageRoute]
+/// for different [TargetPlatform]s.
+///
+/// The [MaterialPageRoute.buildTransitions] method looks up the current
+/// current [PageTransitionsTheme] with `Theme.of(context).pageTransitionTheme`
+/// and delegates to [buildTransitions].
+///
+/// If a builder with a matching platform is not found, the first builder
+/// with a null [PageTransitionsBuilder.platform] is used.
+@immutable
+class PageTransitionsTheme extends Diagnosticable {
+  const PageTransitionsTheme({
+    this.builders = const <PageTransitionsBuilder>[
+      GenericPageTransitionsBuilder(),
+      CupertinoPageTransitionsBuilder(),
+    ],
+  }) : assert(builders != null);
+
+  final Iterable<PageTransitionsBuilder> builders;
```

Comment:
this seems overly general when the reality is we just care about iOS vs anything else.

Reply:
maybe we care about more than that though? I don't really understand why the PageTransitionsBuilders have explicit platforms, and why the general one is different than the mountain view one.